src/uct/tcp/tcp_iface.c

```diff
@@ -258,16 +248,15 @@ uct_tcp_iface_get_sysfs_path(const char *dev_name, char *path_buffer)
                               dev_name);
 
     status = ucs_sys_readdir(ucs_string_buffer_cstr(&dev_path),
-                             uct_tcp_iface_parse_virtual_dev, &data);
-    if (status != UCS_OK) {
+                             uct_tcp_iface_parse_virtual_dev, &dev_path);
+    if ((status != UCS_OK) && (status != UCS_ERR_CANCELED)) {
```

Comment:
IMO it should be `if ((status != UCS_ERR_CANCELED)`
status==UCS_OK means we did not find a corresponding entry
otherwise, need to check if dev_path is an empty string